area without alien plant species

joatop opened the pull request.

anders-kolstad commented:
Takk @joatop . Regner med at denne da er klar for code review og jeg har tagget @DrMattG  og @audhalbritter - har en av dere mulighet til å se på dette?

joatop commented:
> Takk @joatop . Regner med at denne da er klar for code review og jeg har tagget @DrMattG og @audhalbritter - har en av dere mulighet til å se på dette?

Ja, den er klar for code review. ALIE er en del kortere og enklere enn FUNC men bruker samme fremgangsmåte, så kanskje best å begynne med denne her.

audhalbritter commented on indicators/NO_ALIE_001-004/R/NO_ALIE_001-004.qmd:
```diff
@@ -0,0 +1,821 @@
+
+### 2.2 Original units
+
+Abundance data for alien plant species in ANO is given as % cover within a 250m2 sample circle following the national alien species list (Artsdatabanken 2023).
```
Superscripts can be written m^2^ then it will appear properly in the text.

audhalbritter commented on indicators/NO_ALIE_001-004/R/NO_ALIE_001-004.qmd:
````diff
@@ -0,0 +1,821 @@
+```{r}
+#Sys.setlocale("LC_ALL", "no_NB.utf8") #works with æøå or use "Norwegian"
+# ANO data
+ANO.sp <- st_read("C:/Users/joachim.topper/OneDrive - NINA/work/projects/oekologisk tilstand/Naturovervaking_eksport.gdb",
````
same problem here, I cannot access the data and therefore not render the qmd

anders-kolstad replied:
@joatop, kan du kanskje legge dataene på `P:\41201785_okologisk_tilstand_2022_2023`?

audhalbritter reviewed (commented):
Just some minor comments here. But I have not rendered the document yet.

anders-kolstad commented:
@joatop, before I start, can you please

- [ ] fill out `metadata.xlsx`
- [ ] move data to somewhere I can access it (e.g. P:/)
- [ ] render the HTML

joatop commented:
> @joatop, before I start, can you please
> 
> * [ ]  fill out `metadata.xlsx`
> * [ ]  move data to somewhere I can access it (e.g. P:/)
> * [ ]  render the HTML

- data are moved to P:/41201785_okologisk_tilstand_2022_2023/data
- metadate is filled out now
- rendering aborts with following message: Quitting from lines 134-142 [tbl-meta] (NO_ALIE_001-004.qmd)

anders-kolstad commented:
I will fork @joatop 's repo and make some direct changes that he can then merge into his PR. I will also make a review here in the PR.

anders-kolstad commented:
@joatop , I made a [PR to your repo ](https://github.com/joatop/ecRxiv_aliens/pull/1) where I suggest some changes that will make the qmd renderable. If you chose to accept that PR, then this PR will automatically be updated as well. I will also look at the text and everything else in more detail later, but this is what I could manage now. Great if you could also see if some of my suggestion could help you render [the other PR](https://github.com/NINAnor/ecRxiv/pull/123).

joatop commented:
> @joatop , I made a [PR to your repo ](https://github.com/joatop/ecRxiv_aliens/pull/1) where I suggest some changes that will make the qmd renderable. If you chose to accept that PR, then this PR will automatically be updated as well. I will also look at the text and everything else in more detail later, but this is what I could manage now. Great if you could also see if some of my suggestion could help you render [the other PR](https://github.com/NINAnor/ecRxiv/pull/123).

Done! It renders successfully now. Will try to do the same for functional plant community indicators. Guess it does not need the re-naming (001-004 to 002-005) there, or does it?

anders-kolstad commented:
> > @joatop , I made a [PR to your repo ](https://github.com/joatop/ecRxiv_aliens/pull/1) where I suggest some changes that will make the qmd renderable. If you chose to accept that PR, then this PR will automatically be updated as well. I will also look at the text and everything else in more detail later, but this is what I could manage now. Great if you could also see if some of my suggestion could help you render [the other PR](https://github.com/NINAnor/ecRxiv/pull/123).
> 
> Done! It renders successfully now. Will try to do the same for functional plant community indicators. Guess it does not need the re-naming (001-004 to 002-005) there, or does it?

Great! I will read the rest of the text today. NO_FUNC can keep the ID they have already.

anders-kolstad reviewed (changes requested):
Looking good, but I added some comments there and there for improvements and potential follow up work (e.g. the using gbif data).  One more thing to fix is the aggregation pathway in the lsx file. It should be [scale-truncate-transform] - aggregate. Now is scale - aggregate.

joatop commented on indicators/NO_ALIE_001-004/R/NO_ALIE_001-004.qmd:
````diff
@@ -0,0 +1,821 @@
+```{r}
+#Sys.setlocale("LC_ALL", "no_NB.utf8") #works with æøå or use "Norwegian"
+# ANO data
+ANO.sp <- st_read("P:/41201785_okologisk_tilstand_2022_2023/data/Naturovervaking_eksport.gdb",
````
but I guess the format of the ANO-data in gbif is different from the geodatabase on Mdir's kartkatalog...I can change the code to retrieve the data from there directly though.

anders-kolstad replied:
Yes, the GBIF data is probably structured differently and will require some work to get it looking the same. The current solution is OK.

joatop commented:
> [scale-transform]

I added code for downloading the ANO data from Mdir before uploading into R.
Changed the metadata to include Transform now, I thought Scale had it covered, but it is added to it now. Had the same thought about Truncate...while technically the scale-function would truncate if there were cover values larger than 100 or smaller than 0, in practice this is not a thing in this indicator.

joatop commented on indicators/NO_ALIE_002-005/R/NO_ALIE_002-005.qmd:
```diff
@@ -0,0 +1,814 @@
+
+### Calculating national and regional indices
+
+We can calculate national and region-wise medians and 0.025 and 0.975 quantiles. Calculating means is not recommended since these data do not follow a Gaussian distribution.
```
sure, if that is preferred

joatop commented on indicators/NO_ALIE_002-005/R/NO_ALIE_002-005.qmd:
````diff
@@ -0,0 +1,814 @@
+### Calculating national and regional indices
+
+We can calculate national and region-wise medians and 0.025 and 0.975 quantiles. Calculating means is not recommended since these data do not follow a Gaussian distribution.
+```{r}
````
true, but no time left for that now

joatop commented:
> Looking good, but I added some comments there and there for improvements and potential follow up work (e.g. the using gbif data).
> 
> One more thing to fix is the aggregation pathway in the lsx file. It should be [scale-transform] - aggregate. Now is scale - aggregate. I left truncation out since, even though the normalisation function `scal()` includes truncation, there is or should not be, any values to truncate (alien species cover cant be above 100%; correct if that's wrong).
> 
> I also wonder about the title for the indicator. Area without alien plant species sound like there is a binary variable (presence-absence of aliens), but it is a percentage. The title also sound like it should refrain to a proportion of area withuog alien species, but that's not the case. What about _Alien species_ or _Abundance of alien species_?

@name: the indicator evaluates how much area that is not covered by alien species, maybe the name should then be 'area without alien plant species cover'?

joatop commented:
all comments addressed and also transferred to FUNC script

anders-kolstad commented on indicators/NO_ALIE_001-004/R/NO_ALIE_001-004.qmd:
```diff
@@ -0,0 +1,821 @@
+# calculation of the median and quantiles
+# Norway 2022-2024
+df <- res.ANO.alien[(res.ANO.alien$hovedoekosystem_250m2=='skog' | res.ANO.alien$hovedoekosystem_rute=='Forest') & res.ANO.alien$aar %in% c(2022, 2023, 2024) ,c('ano_flate_id','region','hovedoekosystem_250m2','hovedoekosystem_rute','alien')]
+res <-  quantile(df$alien,c(.025, .5, .975), na.rm=T)
```
Solved, but I suggest modifying the code block. The same definition of quantiles `c(0.25, 0.5, 0.75)` is repeated many times, making it easy to forget to update one of them. One can define it once at the start, like

`myQuantiles <- c(.25, .5, .5)`

and then use that string later, like

`res <-  quantile(df$alien, myQuenatile, na.rm=T)`

joatop replied:
good idea, done!

anders-kolstad commented on indicators/NO_ALIE_002-005/R/NO_ALIE_002-005.qmd:
```diff
@@ -0,0 +1,814 @@
+
+## 1. Summary
+
+The 'area without alien plant species' indicator describes to what degree an ecosystem is affected by alien plant species based on the abundance of alien plant species in monitoring sites. The indicator is one-sided as a lack of alien species obviously is not a problem. The reference condition is defined as an ecosystem without any alien plant species (applies to all types of ecosystems and all r ...
```
This suggestion has not been followed up on. Any specific reason why, or was it forgotten?

anders-kolstad commented on indicators/NO_ALIE_002-005/R/NO_ALIE_002-005.qmd:
```diff
@@ -0,0 +1,814 @@
+
+### 2.3 Additional comments about the dataset
+
+<!--# Text here -->
```
This suggestion has not been followed up on. Any specific reason why, or was it forgotten?

anders-kolstad commented on indicators/NO_ALIE_002-005/R/NO_ALIE_002-005.qmd:
```diff
@@ -0,0 +1,814 @@
+
+#### 2.3.1 Instructions for citing, using and accessing data
+
+<!--# Is the data openly available? If not, how can one access it? What are the key references to the datasets?   -->
```
This suggestion has not been followed up on. Any specific reason why, or was it forgotten?

You ended up using data from Kartkatalogen. This can to be explained here. But also link to the GBIF data, since thats where most of the metadata is.

joatop replied:
have done that now

anders-kolstad commented on indicators/NO_ALIE_002-005/R/NO_ALIE_002-005.qmd:
```diff
@@ -0,0 +1,814 @@
+
+### 3.3 Other standards
+
+None.
```
This suggestion has not been followed up on. Any specific reason why, or was it forgotten?

anders-kolstad commented on indicators/NO_ALIE_002-005/R/NO_ALIE_002-005.qmd:
```diff
@@ -0,0 +1,814 @@
+
+### 4.2 Reference levels
+
+The reference threshold for scaling is set to 100% of the monitoring site area without alien plant species. The scaling thresholds for the limit value between good and reduced ecological condition as well as the minimum value are set on an expert-judgement basis: less than 95% area without alien plant species represents the limit threshold, and 0% area without alien plant species represents the m ...
```
This suggestion has not been followed up on. Any specific reason why, or was it forgotten?

anders-kolstad commented:
> all comments addressed and also transferred to FUNC script

Hi @joatop. There are still some unresolved suggestions. When you've looked at them, can you please also re-render the html.

anders-kolstad commented:
> > Looking good, but I added some comments there and there for improvements and potential follow up work (e.g. the using gbif data).
> > One more thing to fix is the aggregation pathway in the lsx file. It should be [scale-transform] - aggregate. Now is scale - aggregate. I left truncation out since, even though the normalisation function `scal()` includes truncation, there is or should not be, any values to truncate (alien species cover cant be above 100%; correct if that's wrong).
> > I also wonder about the title for the indicator. Area without alien plant species sound like there is a binary variable (presence-absence of aliens), but it is a percentage. The title also sound like it should refrain to a proportion of area withuog alien species, but that's not the case. What about _Alien species_ or _Abundance of alien species_?
> 
> @name: the indicator evaluates how much area that is not covered by alien species, maybe the name should then be 'area without alien plant species cover'?

Since there is a strongly non-linear transformation, with X60=5%, then I would argus teh indicator does not, directly at least, show the area witout alien species, and I would prefer a more general and shorter name, like _Alien species_. This is a suggestion, and the final call is yours.

joatop commented on indicators/NO_ALIE_002-005/R/NO_ALIE_002-005.qmd:
```diff
@@ -0,0 +1,814 @@
+
+## 2. About the underlying data
+
+Here, we evaluate the 'area without alien plant species indicator' based on field sampled plant community data from the Norwegian nature monitoring program ANO (Tingstad et al. 2019).
```
Nice! But I am afraid that also I have reached my hours limit now and don't have time resources remaining to prioritize setting this up.

joatop commented on indicators/NO_ALIE_001-004/R/NO_ALIE_001-004.qmd:
```diff
@@ -0,0 +1,821 @@
+
+## fix NiN-variables
+colnames(ANO.geo)
+colnames(ANO.geo)[42:47] <- c("groeftingsintensitet",
```
I removed the part, since it is not necessary for this indicator anyways

joatop commented on indicators/NO_ALIE_001-004/R/NO_ALIE_001-004.qmd:
```diff
@@ -0,0 +1,821 @@
+unique(as.factor(ANO.geo$hovedtype_rute))
+
+ANO.geo$hovedoekosystem_rute <- ANO.geo$hovedtype_rute
+ANO.geo <- ANO.geo %>% mutate(hovedoekosystem_rute=recode(hovedoekosystem_rute, 
```
yeah, but it needs to be next time

joatop commented on indicators/NO_ALIE_002-005/R/NO_ALIE_002-005.qmd:
```diff
@@ -0,0 +1,814 @@
+
+### 4.1 Reference condition
+
+The reference condition is defined as an ecosystem without any alien plant species.
```
have included this text now

joatop commented:
> > > Looking good, but I added some comments there and there for improvements and potential follow up work (e.g. the using gbif data).
> > > One more thing to fix is the aggregation pathway in the lsx file. It should be [scale-transform] - aggregate. Now is scale - aggregate. I left truncation out since, even though the normalisation function `scal()` includes truncation, there is or should not be, any values to truncate (alien species cover cant be above 100%; correct if that's wrong).
> > > I also wonder about the title for the indicator. Area without alien plant species sound like there is a binary variable (presence-absence of aliens), but it is a percentage. The title also sound like it should refrain to a proportion of area withuog alien species, but that's not the case. What about _Alien species_ or _Abundance of alien species_?
> > 
> > 
> > @name: the indicator evaluates how much area that is not covered by alien species, maybe the name should then be 'area without alien plant species cover'?
> 
> Since there is a strongly non-linear transformation, with X60=5%, then I would argus teh indicator does not, directly at least, show the area witout alien species, and I would prefer a more general and shorter name, like _Alien species_. This is a suggestion, and the final call is yours.

Ok, have called it 'Alien plant species' now :)

anders-kolstad commented:
> hope I have covered all comments/suggestions now, the file is rendered

Fantastic! I made some final touched via a [PR to you repo](https://github.com/joatop/ecRxiv_aliens/pull/2). If you merge this then we are all done.
